homeassistant/components/vacuum/__init__.py

```diff
@@ -77,19 +77,19 @@
 class VacuumEntityFeature(IntFlag):
     """Supported features of the vacuum entity."""
 
-    TURN_ON = 1
-    TURN_OFF = 2
+    TURN_ON = 1  # Deprecated, not supported by StateVacuumEntity
+    TURN_OFF = 2  # Deprecated, not supported by StateVacuumEntity
     PAUSE = 4
     STOP = 8
     RETURN_HOME = 16
     FAN_SPEED = 32
     BATTERY = 64
-    STATUS = 128
+    STATUS = 128  # Deprecated, not supported by StateVacuumEntity
     SEND_COMMAND = 256
     LOCATE = 512
     CLEAN_SPOT = 1024
     MAP = 2048
-    STATE = 4096
+    STATE = 4096  # Must be set by vacuum platforms derived from StateVacuumEntity
     START = 8192
 
 
```

Comment:
can we remove this? cause the deprecated uses of VacuumEntity (demo, mqtt, ecovacs) uses the wrong define ->  VacuumEntityFeature.TURN_ON anyhow

SUPPORT_TURN_ON:
is not used at all

SUPPORT_TURN_OFF:
is used only in test/components/alexa we should remove it since its for new StateVacuumEntity deprecated anyhow

SUPPORT_PAUSE:
used in test/components/alexa/test_smart_home.py -> we should change it to enum VacuumEntityFeature.PAUSE
used in test/components/google/test_trait.py -> we should change it to enum VacuumEntityFeature.PAUSE

SUPPORT_STOP:
used in test/components/alexa/test_smart_home.py -> we should change it to enum VacuumEntityFeature.SUPPORT_STOP
used in test/components/google/test_trait.py -> we should change it to enum VacuumEntityFeature.SUPPORT_STOP

SUPPORT_RETURN_HOME
used in test/components/alexa/test_smart_home.py -> we should change it to enum VacuumEntityFeature.RETURN_HOME
used in test/components/homekit/test_get_accessories.py -> we should change it to enum VacuumEntityFeature.RETURN_HOME

SUPPORT_FAN_SPEED
used in test/components/alexa/test_smart_home.py -> we should change it to enum VacuumEntityFeature.FAN_SPEED

SUPPORT_BATTERY
used in components/google_assistant/traits.py -> we should change it to enum VacuumEntityFeature.SUPPORT_BATTERY
used in test/components/google/test_trait.py -> we should change it to enum VacuumEntityFeature.SUPPORT_BATTERY

SUPPORT_STATUS -> not used at all
SUPPORT_SEND_COMMAND -> not used at all

SUPPORT_LOCATE
used in components/google_assistant/traits.py -> we should change it to enum VacuumEntityFeature.SUPPORT_LOCATE
used in test/components/google/test_trait.py -> we should change it to enum VacuumEntityFeature.SUPPORT_LOCATE

SUPPORT_CLEAN_SPOT -> not used at all
SUPPORT_MAP -> not used at all
SUPPORT_STATE -> not used at all

SUPPORT_START
used in test/components/alexa/test_smart_home.py -> we should change it to enum VacuumEntityFeature.START
used in test/components/homekit/test_get_accessories.py -> we should change it to enum VacuumEntityFeature.SUPPORT_START

can prepare a merge if you like?

Reply:
Sure, please prepare a PR cleaning up the use of the deprecated constants

Reply:
created: https://github.com/home-assistant/core/pull/96202